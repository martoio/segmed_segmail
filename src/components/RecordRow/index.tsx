import * as React from 'react';

import Record from '../../models/record';

interface RecordRowProps {
    record: Record;
    onClick: (e: React.MouseEvent) => void;
}

const DEFAULT_BODY_LENGTH = 140;

export const RecordRow = (props: RecordRowProps) => {
    const { record } = props;
    return (
        <tr
            onClick={props.onClick}
        >
            <td>{record.id}</td>
            <td>{record.title}</td>
            <td>{record.body.substring(0, DEFAULT_BODY_LENGTH)}...</td>
        </tr>
    );
};

export default RecordRow;