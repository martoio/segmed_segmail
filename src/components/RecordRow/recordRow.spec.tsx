import * as React from 'react';
import {shallow, configure, render} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import RecordRow from './';
import Record from '../../models/record';

configure({ adapter: new Adapter() });

const record = new Record(1, 'dummy', 'test');

describe('RecordRow component', () => {
    it('should render correctly', () => {
        const tree = shallow(<RecordRow record={record} onClick={() => {}} />);

        expect(tree).toMatchSnapshot();
    });
})
