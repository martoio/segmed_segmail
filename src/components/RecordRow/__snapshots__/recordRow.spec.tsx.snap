// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordRow component should render correctly 1`] = `
<tr
  onClick={[Function]}
>
  <td>
    1
  </td>
  <td>
    dummy
  </td>
  <td>
    test
    ...
  </td>
</tr>
`;
